src/rgw/rgw-orphan-list

```diff
@@ -57,19 +58,49 @@ echo "Pool is \"$pool\"."
 echo "Note: output files produced will be tagged with the current timestamp -- ${timestamp}."
 
 echo "running 'rados ls' at $(date)"
-rados ls --pool="$pool" --format=json-pretty --all >"$rados_out" 2>"$rados_err"
+# since --format is not specified, plain should be used
+rados ls --pool="$pool" --all >"$rados_out" 2>"$rados_err"
 if [ "$?" -ne 0 ] ;then
     error_out "rados ls" "$rados_err"
 fi
 
-# check for namespaces and error out if any found
-grep '^[[:blank:]]*"namespace":' "$rados_out" | grep --silent '[^:]*: "[^"]'
-if [ "${PIPESTATUS[1]}" -eq 0 ] ;then
-    error_out "rados ls" "$rados_out" "Found one or more RADOS objects existing in a namespace."
+# NOTE: Each entry (line of output) of `rados ls --all` should be in
+# one of four formats depending on whether or not an entry has a
+# namespace and/or locator:
+#
+#   <TAB>oid
+#   <TAB>oid<TAB>locator
+#   namespace<TAB>oid
+#   namespace<TAB>oid<TAB>locator
+#
+# Any occurrence of the 3rd or 4th (i.e., existence of namespace)
+# should cause this script to error out. Any occurence of the 2nd
+# (i.e., existence of locator) should generate a warning but allow the
+# script to continue to the end. We expect all entries to be of the
+# first line format.
+
+# check for namespaces and error out if any found; any line that
+# begins with graphical characters indicates a namespace as those
+# without a namespace will start with a tab
+grep '^[[:graph:]]' "$rados_out" >"$rados_odd"
```

Comment:
Using this rather than explicitly tab bothers me slightly.

grep "$(echo -n '^\t')" seems to work under dash, so I think it might be portable. May not be worth worrying about, since on RGW we control how things are named.

Reply:
Just so I understand, rather than looking for a "graph" character you think I should look for a not-tab at the start of the line, as in `grep '^[^\t]' ...`?

Reply:
Yes. Or grep -v for a line starting with a tab.

But it may not be an issue if the OIDs for RGW just won't start with blankspace.

Reply:
Thanks, @adamemerson! I liked your suggestions and modified the regular expressions.